### ISSUE TITLE
MeshInstance3D doesn't update when material textures are changed (albedo, normal, …)

### ISSUE THREAD

**Issue by floppyhammer** [godotengine/godot#48886]
**Godot version:**
v4.0.dev.custom_build.c340ed639

**OS/device including version:**
Linux, Vulkan

**Issue description:**
MeshInstance3D doesn't update when material textures (`albedo_texture`, `emission_texture`, etc.) are changed. They only take effect after some other material properties are changed or after the scene is reloaded.
![Peek 2021-05-20 19-28](https://user-images.githubusercontent.com/28705694/118971712-55304780-b9a2-11eb-8863-52e6377d50f1.gif)

**Steps to reproduce:**
Add a MeshInstance3D node, add a mesh, add a StandardMaterial3D, attach a texture to `albedo_texture`.

**Minimal reproduction project:**
[test.zip](https://github.com/godotengine/godot/files/6515511/test.zip)

**Comment by Calinou** [comment]
This may be a duplicate of another issue, but I can't find it right now.

(For the record, I can reproduce this on commit https://github.com/godotengine/godot/commit/f2d55f3c0fabcd4ba4c14d56a6abd4627ebafc4a. As a workaround, you can save the scene then use **Scene > Revert Scene**.)

**Comment by Valeryn4** [comment]
```
void BaseMaterial3D::_update_shader() {
	dirty_materials->remove(&element);

	MaterialKey mk = _compute_key();
	if (mk == current_key) {
		return; //no update required in the end
	}
```

The MaterialKey does not contain any information about updating the texture. Therefore, the conditions are always true.

**Comment by Valeryn4** [comment]
[There is a variant of such a patch.](https://github.com/Valeryn4/godot/commit/0bc97afc648fdc585c8a5bfbb4ad9ce62d1addfa)
It is necessary to store information that the texture was changed before the call to "_update"
It looks like a crutch = (

**Comment by floppyhammer** [comment]
@Valeryn4 It might not be a good idea to update shader whenever some texture uniform changes. I suppose the problem is more related to texture cache.

**Comment by Valeryn4** [comment]
I know.
I pointed out that this is just a patch.
If someone has time to do better, do it)

**Comment by floppyhammer** [comment]
Fixed by #50229.
